homeassistant/components/smartthings/media_player.py

```diff
@@ -100,27 +99,25 @@ def __init__(self, client: SmartThings, device: FullDevice) -> None:
         )
 
     def _determine_features(self) -> MediaPlayerEntityFeature:
-        flags = MediaPlayerEntityFeature(0)
-        playback_commands = self.get_attribute_value(
-            Capability.MEDIA_PLAYBACK, Attribute.SUPPORTED_PLAYBACK_COMMANDS
+        flags = (
+            MediaPlayerEntityFeature.VOLUME_SET
+            | MediaPlayerEntityFeature.VOLUME_STEP
+            | MediaPlayerEntityFeature.VOLUME_MUTE
         )
-        if "play" in playback_commands:
-            flags |= MediaPlayerEntityFeature.PLAY
-        if "pause" in playback_commands:
-            flags |= MediaPlayerEntityFeature.PAUSE
-        if "stop" in playback_commands:
-            flags |= MediaPlayerEntityFeature.STOP
-        if "rewind" in playback_commands:
-            flags |= MediaPlayerEntityFeature.PREVIOUS_TRACK
-        if "fastForward" in playback_commands:
-            flags |= MediaPlayerEntityFeature.NEXT_TRACK
-        if self.supports_capability(Capability.AUDIO_VOLUME):
-            flags |= (
-                MediaPlayerEntityFeature.VOLUME_SET
-                | MediaPlayerEntityFeature.VOLUME_STEP
+        if self.supports_capability(Capability.MEDIA_PLAYBACK):
+            playback_commands = self.get_attribute_value(
+                Capability.MEDIA_PLAYBACK, Attribute.SUPPORTED_PLAYBACK_COMMANDS
             )
-        if self.supports_capability(Capability.AUDIO_MUTE):
-            flags |= MediaPlayerEntityFeature.VOLUME_MUTE
+            if "play" in playback_commands:
+                flags |= MediaPlayerEntityFeature.PLAY
+            if "pause" in playback_commands:
+                flags |= MediaPlayerEntityFeature.PAUSE
+            if "stop" in playback_commands:
+                flags |= MediaPlayerEntityFeature.STOP
+            if "rewind" in playback_commands:
+                flags |= MediaPlayerEntityFeature.PREVIOUS_TRACK
+            if "fastForward" in playback_commands:
+                flags |= MediaPlayerEntityFeature.NEXT_TRACK
```

Comment:
Would be nicer to change this to use a mapping and loop over the playback commands and map them to HA features

Reply:
Oh that's a good idea